src/codecoverage/bot/code_coverage_bot/zero_coverage.py

```diff
@@ -85,7 +86,51 @@ def get_fileinfo(self, filenames):
 
         return res
 
-    def generate(self, artifacts, hgrev, gitrev, out_dir='.'):
+    def generate_zero_coverage_some_platform_only(self, report, hgrev, gitrev, out_dir='.'):
+        '''
+        :param report: dictionary with key platform name, and value is report generated
+                       using ZeroCov.generate for that platform
+        :param hgrev: hg revision, must be same for all reports
+        :param gitrev: git revision, must be same for all reports
+        :param out_dir: directory to output the json result
+        '''
+
+        platforms = ['linux', 'windows']
+
+        git_revision = set([report[platform]['github_revision'] for platform in platforms])
+        hg_revision = set([report[platform]['hg_revision'] for platform in platforms])
+        if len(git_revision) != 1 and len(hg_revision) != 1:
```

Comment:
Nit: turn this into an actual assertion:
`assert len(git_revision) == 1 and len(hg_revision) == 1, 'git revision and mercurial revision must be the same'`